pending/19-one-page-apps.pod

```diff
@@ -0,0 +1,95 @@
+=head1 Making a one-page app using Dancer2::Plugin::Ajax
+
+It's possible to make bits of DOM on a Dancer2 template behave like a one-page application, auto-updating
+themselves using some clever JavaScript and L<Dancer2::Plugin::Ajax>. In this article, I'll give you a quick
+runthrough of how to do that.
+
+=head2 The example: messaging
+
+In this example, I want the app to check every few minutes for messages coming from some other system. I'll
+have Dancer2 look for the actual messages, and the frontend of the app will call a route using C<.ajax>, and 
+if there's a message, update some DOM. I'll be using jQuery for this, though other tookits have similar sorts
+of constructs.
+
+Here's the HTML needed for the message block:
+
+=begin html
+
+<div id='messageBlock' style='min-height: 50px; padding:10px; margin: 20px; border: 1px solid black'>
+  <h2>Messages from Beyond:</h2>
+  <div id='messageResults'></div>    <!-- Nothing in this div.  Yet. -->
+</div>
+
+=end
+
+And here's the route we'll be calling. In this example, I'm randomly generating the messages, but in a real
+app, you might be fetching them from a message box, a file, a database, or whatever. Because it's what I'm used
+to, I set the application-type in my config.yml to C<'application/json'>, but this can do xml, or whatever
+you need; just make sure that's what you're returning!
+
+=begin perl
+
+use Dancer2::Plugin::Ajax;
+
+ajax '/messages' => sub {
+   my @messages = (
+      'Boo! I betcha I scared you, ha-ha!',
+      'A girl has no name, but she still has a list.',
+      'Space, the final frontier.',
+      '',
+      '<h1>Really big text</h1>',
+   );
+   return to_json({ message => @messages[int(rand(5))] });
```

Comment:
Alternatively, you could write this as `send_as JSON => { message => @messages[int(rand(5))] };`

Reply:
I've made that change!